subsys/mpsl/Kconfig

```diff
@@ -25,6 +25,108 @@ config MPSL_TIMESLOT_SESSION_COUNT
 	help
 	  Maximum number of timeslot sessions.
 
+DT_COMPAT_NORDIC_NRF21540_GPIO := nordic,nrf21540-fem
+DT_COMPAT_SKYWORKS_SKY66112_11 := skyworks,sky66112-11
+
+config CAN_USE_MPSL_FEM_NRF21540_GPIO
+	bool
+	default $(dt_nodelabel_has_compat,nrf_radio_fem,$(DT_COMPAT_NORDIC_NRF21540_GPIO)) && !BT
+
+config CAN_USE_MPSL_FEM_SKY66112_11
+	bool
+	default $(dt_nodelabel_has_compat,nrf_radio_fem,$(DT_COMPAT_SKYWORKS_SKY66112_11))
+
+config MPSL_FEM
+	bool "Radio front-end module (FEM) support"
+# MPSL FEM is not supported on 53 yet
+	depends on !(SOC_NRF5340_CPUAPP || SOC_NRF5340_CPUNET)
+	default CAN_USE_MPSL_FEM_NRF21540_GPIO || CAN_USE_MPSL_FEM_SKY66112_11
```

Comment:
@joerchan , I would like to understand the reasons to remove default.

If nRF21540 is present in the design, it must be enabled in software. If it is disabled, radio protocols are unusable, it would work as an attenuator.

I thought that options in Kconfig should depend on devicetree description.
1. If nRF21540 is missing, it cannot be enabled in Kconfig.
2. If nRF21540 GPIO interface is defined, it is the only available option in Kconfig (and default one).
3. If nRF21540 SPI interface is defined, it is the only one available option in Kconfig
4. If both GPIO and SPI interfaces are defined there are three options in Kconfig (SPI is default):
    1. SPI
    2. GPIO
    3. Mixed (SPI for power, GPIO for timings)

Currently only GPIO binding is defined (and implemented). Because of that I was expecting support for points 1 and 2 from the list above in this PR.

What are your opinions?

Reply:
@hubertmis I meant that the default selected based on the support. Because the support should go with depends instead as they are now.

Having `default y` is fine.

Reply:
@joerchan , do I understand correctly that you suggest something like:
```
config MPSL_FEM
	bool "Radio front-end module (FEM) support"
	depends on !(SOC_NRF5340_CPUAPP || SOC_NRF5340_CPUNET) && \
	           (MPSL_FEM_NRF21540_GPIO_SUPPORT || MPSL_FEM_SKY66112_11_SUPPORT)
        default y

choice MPSL_FEM_CHOICE
	prompt "Radio front-end module (FEM) type"

#This new option will be added after SPI support is implemented
config MPSL_FEM_NRF21540_SPI
	depends on MPSL_FEM_NRF21540_SPI_SUPPORT
	bool "nRF21540 front-end module in SPI mode"
        default y

config MPSL_FEM_NRF21540_GPIO
	depends on MPSL_FEM_NRF21540_GPIO_SUPPORT
	bool "nRF21540 front-end module in GPIO mode"
        default y if !MPSL_FEM_NRF21540_SPI_SUPPORT # To be added after SPI support is added
	help
	  Note: nRF21540 is unavailable when Bluetooth LE is enabled

#This new option will be added after SPI support is implemented
config MPSL_FEM_NRF21540_MIXED
	depends on MPSL_FEM_NRF21540_SPI_SUPPORT && MPSL_FEM_NRF21540_GPIO_SUPPORT
	bool "nRF21540 front-end module in mixed mode"
	help
	  Mixed mode means that SPI is used to control TX power, while GPIO is used to control radio timings. This mode should not be used for low-power applications, because nRF21540 is powered on all the time to keep TX power settings in its ram. For low-power applications use MPSL_FEM_NRF21540_SPI instead.

config MPSL_FEM_SKY66112_11
	depends on MPSL_FEM_SKY66112_11_SUPPORT
	bool "SKY66112-11 front-end module"
```

Reply:
@hubertmis As i mentioned in another of the comments the order of choices determines the defaults, so it is better to order them in preferred order to avoid putting in too many default statements with condititions.

So like this I think would be best:
```
config MPSL_FEM
	bool "Radio front-end module (FEM) support"
	depends on !(SOC_NRF5340_CPUAPP || SOC_NRF5340_CPUNET) && \
	           (MPSL_FEM_NRF21540_GPIO_SUPPORT || MPSL_FEM_SKY66112_11_SUPPORT)
        default y

choice MPSL_FEM_CHOICE
	prompt "Radio front-end module (FEM) type"

#This new option will be added after SPI support is implemented
config MPSL_FEM_NRF21540_SPI
	depends on MPSL_FEM_NRF21540_SPI_SUPPORT
	bool "nRF21540 front-end module in SPI mode"

config MPSL_FEM_NRF21540_GPIO
	depends on MPSL_FEM_NRF21540_GPIO_SUPPORT
	bool "nRF21540 front-end module in GPIO mode"
	help
	  Note: nRF21540 is unavailable when Bluetooth LE is enabled

#This new option will be added after SPI support is implemented
config MPSL_FEM_NRF21540_MIXED
	depends on MPSL_FEM_NRF21540_SPI_SUPPORT && MPSL_FEM_NRF21540_GPIO_SUPPORT
	bool "nRF21540 front-end module in mixed mode"
	help
	  Mixed mode means that SPI is used to control TX power, while GPIO is used to control radio timings. This mode should not be used for low-power applications, because nRF21540 is powered on all the time to keep TX power settings in its ram. For low-power applications use MPSL_FEM_NRF21540_SPI instead.

config MPSL_FEM_SKY66112_11
	depends on MPSL_FEM_SKY66112_11_SUPPORT
	bool "SKY66112-11 front-end module"
```